src/intercept.c

```diff
@@ -675,7 +694,7 @@ intercept_routine(struct context *context)
 		 * the clone_child_intercept_routine instead, executing
 		 * it on the new child threads stack, then returns to libc.
 		 */
-		if (desc.nr == SYS_clone && desc.args[1] != 0)
+		if (is_clone_syscall(desc.nr) && desc.args[1] != 0)
```

Comment:
The arguments are not them same here. `desc.args[1]` is the stack pointer in `clone`. The stack pointer for clone3 is `desc.args[0]->stack`. 

```
#ifdef SYS_clone3
else if(desc.nr == SYS_clone3 && ((struct clone_args*) desc.args[0])->stack != 0) {
    return (struct wrapper_ret){
				.rax = context->rax, .rdx = 2 };
}
#endif
```